lib/scripts/users_list.rb

```diff
@@ -0,0 +1,32 @@
+require 'csv'
+
+class UsersList
+  attr_accessor :csv_file
+
+  def initialize
+    @csv_file = CSV.open(File.join(Rails.root, "/lib/scripts/results/users_list.csv"), "w") do |csv|
+      csv << ['Formal Name', 'netID', 'email', 'Current sign in', 'Last sign in']
+    end
+  end
+
+  def populate_user_info
+    if self.csv_file.closed?
+      self.csv_file.reopen(self.csv_file.path, "a")
+    end
```

Comment:
The csv file actually gets closed after initializing. I pass a block to it to add headers. But, I think you are right that it would be better to remove this `closed?` branch, and remove the block on initialize and just add the headers outside of the block.

Reply:
aah ruby opens the file and closes it for you at the end of the block. Still go with @Meowcenary comments above: opening and closing just in the `populate_user_info`.